src/object/schema_manager.c

```diff
@@ -15513,7 +15513,16 @@ sm_truncate_using_delete (MOP class_mop)
   /* We will run a DELETE statement with triggers disabled. */
   save_tr_state = tr_set_execution_state (false);
 
-  (void) snprintf (delete_query, sizeof (delete_query), "DELETE /*+ RECOMPILE */ FROM [%s];", class_name);
+  if (prm_get_integer_value (PRM_ID_SUPPLEMENTAL_LOG) > 0)
+    {
+      (void) snprintf (delete_query, sizeof (delete_query), "DELETE /*+ RECOMPILE NO_SUPPLEMENTAL_LOG */ FROM [%s];",
+		       class_name);
+    }
+  else
+    {
+      (void) snprintf (delete_query, sizeof (delete_query), "DELETE /*+ RECOMPILE */ FROM [%s];", class_name);
```

Comment:
How about always using the hint in this sort of context (DDL)? The value of `PRM_ID_SUPPLEMENTAL_LOG` will be checked in the code later where the supplement log is appended. 
```suggestion
    
```

Reply:
OK, It may reduce the cost of checking conditions.